### PR TITLE
Enable uninstallation and add support for CodePeer

### PR DESCRIPTION
**Pull request by treiher**

### CODE REVIEW

File: lmod_manager/lmod_manager.py [hunk 1]
```diff
@@ -126,5 +126,125 @@ def install(args: argparse.Namespace) -> Union[int, str]:
...
+
+
+def _installation_file(module_type: Type) -> Path:
+    if module_type is Type.GNAT:
+        return Path("bin/gnat")
+    if module_type is Type.SPARK:
+        return Path("bin/gnatprove")
+    if module_type is Type.CODEPEER:
+        return Path("bin/codepeer")
+    assert_never(module_type)  # pragma: no cover
```
Comment (jklmnn) [inline lmod_manager/lmod_manager.py]: I've seen multiple of these if clauses, maybe it makes sense to create an abstract type class with properties that are the implemented for GNAT, SPARK and Codepeer?

**Review by jklmnn — CHANGES_REQUESTED** [review]

File: lmod_manager/lmod_manager.py (hunk 1 above)
Comment (treiher) [reply lmod_manager/lmod_manager.py]: You are right, I have refactored the code.

**Review by jklmnn — DISMISSED** [review]

File: lmod_manager/lmod_manager.py [hunk 2]
```diff
@@ -7,124 +7,270 @@
...
+        shutil.rmtree(self._extracted_archive_dir())
+
+        with open(config_file, "w", encoding="utf-8") as f:
+            f.write(
+                f"""\
+local pkgName = myModuleName()
+local version = myModuleVersion()
+local pkg     = pathJoin("{installation_dir}",pkgName,version,"bin")
+prepend_path("PATH", pkg)
+"""
```
Comment (senier) [inline lmod_manager/lmod_manager.py]: You could use [`testwrap.dedent`](https://docs.python.org/3.6/library/textwrap.html#textwrap.dedent) to make this a bit more readable.

**Review by senier — CHANGES_REQUESTED** [review]

**Review by jklmnn — APPROVED** [review]

File: lmod_manager/lmod_manager.py (hunk 2 above)
Comment (treiher) [reply lmod_manager/lmod_manager.py]: Changed.

**Review by senier — APPROVED** [review]

**Pull request merged**